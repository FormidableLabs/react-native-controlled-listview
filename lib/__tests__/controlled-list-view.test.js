/*global it, expect*/
/*eslint-disable react/no-multi-comp*/
import { Text } from 'react-native';
import React from 'react';
import ControlledListView from '../controlled-list-view';
import people from './__data__/people';
import renderer from 'react-test-renderer';

const renderRow = (person) => (
  <Text>{`${person.lastName}, ${person.firstName}`}</Text>
);

const renderSectionHeader = (sectionData, initial) => (
  <Text>{initial}</Text>
);

// helper to update component props
const setProps = (component, changedProps) => {
  const instance = component.getInstance();
  const nextProps = {
    ...instance.props,
    ...changedProps
  };
  instance.componentWillReceiveProps(nextProps);
  instance.props = nextProps;
};

it('updates list when items prop is updated', () => {
  const component = renderer.create(
    <ControlledListView items={people} renderRow={renderRow} />
  );
  expect(component.toJSON()).toMatchSnapshot();

  setProps(component, { items: [people[0]] });
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders items in correct sort order', () => {
  const tree = renderer.create(
    <ControlledListView
      items={people}
      sortBy={(a, b) => a.lastName < b.lastName}
      renderRow={renderRow}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('splits items into sections', () => {
  const tree = renderer.create(
    <ControlledListView
      items={people}
      sortBy={(a, b) => a.lastName < b.lastName}
      sectionBy={(person) => person.lastName[0]}
      renderRow={renderRow}
      renderSectionHeader={renderSectionHeader}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
