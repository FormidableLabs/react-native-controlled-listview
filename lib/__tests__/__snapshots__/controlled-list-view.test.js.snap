exports[`test renders items in correct sort order 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 6,
    }
  }
  dataSourceShouldUpdate={[Function]}
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  rowHasChanged={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  sectionHeaderHasChanged={[Function]}
  stickyHeaderIndices={Array []}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Bonk, Mildred
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Gately, Don
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Hal
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Avril
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Orin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Pemulis, Michael
  </Text>
</ScrollView>
`;

exports[`test splits items into sections 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 6,
    }
  }
  dataSourceShouldUpdate={[Function]}
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  renderSectionHeader={[Function]}
  rowHasChanged={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  sectionHeaderHasChanged={[Function]}
  stickyHeaderIndices={
    Array [
      0,
      2,
      4,
      8,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    B
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Bonk, Mildred
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    G
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Gately, Don
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    I
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Orin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Avril
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Hal
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    P
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Pemulis, Michael
  </Text>
</ScrollView>
`;

exports[`test updates list when items prop is updated 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 6,
    }
  }
  dataSourceShouldUpdate={[Function]}
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  rowHasChanged={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  sectionHeaderHasChanged={[Function]}
  stickyHeaderIndices={Array []}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Orin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Gately, Don
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Avril
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Pemulis, Michael
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Hal
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Bonk, Mildred
  </Text>
</ScrollView>
`;

exports[`test updates list when items prop is updated 2`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 1,
    }
  }
  dataSourceShouldUpdate={[Function]}
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  rowHasChanged={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  sectionHeaderHasChanged={[Function]}
  stickyHeaderIndices={Array []}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Incandenza, Orin
  </Text>
</ScrollView>
`;
